src/CMakeLists.txt

```diff
@@ -98,16 +102,16 @@ add_library(${LIBRARY_NAME} SHARED ${LIBRARY_SOURCES} ${LIBRARY_PUBLIC_HEADERS})
 add_library(${PROJECT_NAME}::${LIBRARY_NAME} ALIAS ${LIBRARY_NAME})
 target_link_libraries(${LIBRARY_NAME} PUBLIC OpenSSL::Crypto OpenSSL::SSL Boost::boost)
 
-if(HSM_ENABLED)
-  target_compile_definitions(${LIBRARY_NAME} PUBLIC HSM_ENABLED)
-endif()
-
 if(DILITHIUM_ENABLED)
     target_link_libraries(${LIBRARY_NAME} PRIVATE dilithium2_ref dilithium3_ref dilithium5_ref aes256ctr_ref fips202_ref)
 endif()
 
+#Include BINARY_DIR so that sources can find the config header.
+target_include_directories(${LIBRARY_NAME} PRIVATE ${CMAKE_BINARY_DIR})
+
 target_include_directories(${LIBRARY_NAME} INTERFACE
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
```

Comment:
Doesn't this just repeat the definition of l. 110?
It should work without it. Otherwise, it will not work with the generated exported target.